media-api/app/controllers/MediaApi.scala

```diff
@@ -278,14 +294,46 @@ object MediaApi extends Controller with ArgoHelpers {
     }
   }
 
+  private def getRelatedLabelsLink(mainLabel: String, selectedLabels: List[String] = Nil) = {
+    val uriTemplate = URITemplate(s"$rootUri/suggest/edits/labels/{label}/sibling-labels{?selectedLabels}")
+    val paramMap = Map(
+      "label" -> Some(mainLabel),
+      "selectedLabels" -> Some(selectedLabels.mkString(",")).filter(_.trim.nonEmpty)
+    )
+    val paramVars = paramMap.map { case (key, value) => key := value }.toSeq
+    val uri = uriTemplate expand (paramVars: _*)
+
+    Link("related-labels", uri)
+  }
+
   def suggestMetadataCredit(q: Option[String], size: Option[Int]) = Authenticated.async { request =>
     ElasticSearch
       .completionSuggestion("suggestMetadataCredit", q.getOrElse(""), size.getOrElse(10))
       .map(c => respondCollection(c.results))
   }
 
-  def suggestLabelSiblings(label: String) = Authenticated.async { request =>
-    ElasticSearch.labelSiblingsSearch(label) map aggregateResponse
+  def suggestLabelSiblings(label: String, selectedLabels: Option[String]) = Authenticated.async { request =>
```

Comment:
Not really - you can set that to what ever you want here from what ever client you want.

As the [Link is created in the API](https://github.com/guardian/grid/pull/1315/files#diff-56cb21a41bb269d77a616e1f7a4f42e9R241) we aren't tying any state to anything really.

We set the `selectedLabels` on the API, create the link and send it to this endpoint.
Any other ideas on where to set this?